rmw_cyclonedds_cpp/CMakeLists.txt

```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.15)
```

Comment:
```suggestion
cmake_minimum_required(VERSION 3.20)
```

Reply:
yeah it would be probably better to upgrade `3.20`, see https://github.com/ros2/demos/pull/714#pullrequestreview-2740541209